SimTracker/TrackerHitAssociation/test/BuildFile.xml

```diff
@@ -3,6 +3,9 @@
 <use   name="DataFormats/Common"/>
 <use   name="FWCore/ParameterSet"/>
 <use   name="FWCore/Utilities"/>
+<use name="cuda"/>
+<use name="HeterogeneousCore/CUDACore"/>
+<use name="HeterogeneousCore/Product"/>
```

Comment:
Is the `HeterogeneousCore/Product` really needed? (if yes, we should understand why and eventually work around it)

Reply:
```
>> Building edm plugin tmp/slc7_amd64_gcc820/src/SimTracker/TrackerHitAssociation/test/SimTrackerTrackerHitAssociationTestModules/libSimTrackerTrackerHitAssociationTestModules.so
/home/cmssw/slc7_amd64_gcc820/external/gcc/8.2.0-pafccj/bin/../lib/gcc/x86_64-unknown-linux-gnu/8.3.1/../../../../x86_64-unknown-linux-gnu/bin/ld: tmp/slc7_amd64_gcc820/src/SimTracker/TrackerHitAssociation/test/SimTrackerTrackerHitAssociationTestModules/ClusterTPCUDAdump.cc.o: in function `ClusterTPCUDAdump::analyze(edm::StreamID, edm::Event const&, edm::EventSetup const&) const':
ClusterTPCUDAdump.cc:(.text+0x270): undefined reference to `impl::CUDAScopedContextBase::CUDAScopedContextBase(CUDAProductBase const&)'
/home/cmssw/slc7_amd64_gcc820/external/gcc/8.2.0-pafccj/bin/../lib/gcc/x86_64-unknown-linux-gnu/8.3.1/../../../../x86_64-unknown-linux-gnu/bin/ld: ClusterTPCUDAdump.cc:(.text+0x2a2): undefined reference to `impl::CUDAScopedContextGetterBase::synchronizeStreams(int, CUstream_st*, bool, CUevent_st*)'
/home/cmssw/slc7_amd64_gcc820/external/gcc/8.2.0-pafccj/bin/../lib/gcc/x86_64-unknown-linux-gnu/8.3.1/../../../../x86_64-unknown-linux-gnu/bin/ld: ClusterTPCUDAdump.cc:(.text+0x2b1): undefined reference to `CUDAScopedContextProduce::~CUDAScopedContextProduce()'
/home/cmssw/slc7_amd64_gcc820/external/gcc/8.2.0-pafccj/bin/../lib/gcc/x86_64-unknown-linux-gnu/8.3.1/../../../../x86_64-unknown-linux-gnu/bin/ld: tmp/slc7_amd64_gcc820/src/SimTracker/TrackerHitAssociation/test/SimTrackerTrackerHitAssociationTestModules/ClusterTPCUDAdump.cc.o: in function `ClusterTPCUDAdump::analyze(edm::StreamID, edm::Event const&, edm::EventSetup const&) const [clone .cold.103]':
ClusterTPCUDAdump.cc:(.text.unlikely+0x56): undefined reference to `CUDAScopedContextProduce::~CUDAScopedContextProduce()'
collect2: error: ld returned 1 exit status
```